cmd/otel-allocator/allocation/consistent_hashing.go

```diff
@@ -0,0 +1,194 @@
+package allocation
+
+import (
+	"fmt"
+	"github.com/buraksezer/consistent"
+	"github.com/cespare/xxhash/v2"
+	"github.com/go-logr/logr"
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/diff"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/url"
+	"sync"
+)
+
+var _ Allocator = &consistentHashingAllocator{}
+
+const consistentHashingStrategyName = "consistent-hashing"
+
+type hasher struct{}
+
+func (h hasher) Sum64(data []byte) uint64 {
+	return xxhash.Sum64(data)
+}
+
+type consistentHashingAllocator struct {
+	// m protects consistentHasher, collectors and targetItems for concurrent use.
+	m sync.RWMutex
+
+	consistentHasher *consistent.Consistent
+
+	// collectors is a map from a Collector's name to a Collector instance
+	collectors map[string]*Collector
+
+	// targetItems is a map from a target item's hash to the target items allocated state
+	targetItems map[string]*TargetItem
+
+	log logr.Logger
+}
+
+func newConsistentHashingAllocator(log logr.Logger) Allocator {
+	config := consistent.Config{
+		PartitionCount:    1061,
+		ReplicationFactor: 5,
+		Load:              1.1,
```

Comment:
I'm not sure... I think i'm going to open a follow up issue to make these configurable as it would require some refactoring of how configuration is passed down to the allocation strategies as right now it's only the string. Is that alright?

Reply:
Yeah, that's fine.  I think using the string for configuration might be fine, as long as it's well specified.  Something like URL query params style might work:

```
allocationStrategy: "consistentHashing?load=1.5&partitionCount=3000"
```

That would make it a straightforward addition while leaving plenty of flexibility.  Don't need to solve now though, can figure it out on the follow up issue.